proposals/lip-0052.md

```diff
@@ -706,7 +708,8 @@ def execute(
             Fee.payFee(FEE_CREATE_NFT)
             createNFTEntry(recipientAddress, nftID, receivedAttributes)
             createUserEntry(recipientAddress, nftID)
-        else: # return the NFT to the sender
+        else: # Return the NFT to the sender
+            Fee.payFee(FEE_CREATE_NFT)
```

Comment:
We were discussing this case with Alessandro and I think this line could be executed only in one case: the NFT is bounced on the native chain and has to return back to the sending chain. This should not normally happen as the native chain of an NFT should always accept it.

To be a bit more strict and handle this case better we could add the following check to the ccm verification:

```
if getChainID(nftID) != OWN_CHAIN_ID and (ccm.status == CCM_STATUS_CODE_MODULE_NOT_SUPPORTED or ccm.status == CCM_STATUS_CODE_CROSS_CHAIN_COMMAND_NOT_SUPPORTED):
    raise Exception()
```
This will guarantee that if a native chain of an NFT suddenly denies an incoming NFT it is terminated. 
I think this measure together with the `Fee.payFee(FEE_CREATE_NFT)` prevents malicious actors from sending NFTs for free.

Reply:
I understand your argument, but I think the `if getChainID(nftID) != OWN_CHAIN_ID` is not necessary since this part of code is already under [this `else`](https://github.com/LiskHQ/lips/blob/ca1bd3e68aaf7326de93983506eacd47f483aafa/proposals/lip-0052.md?plain=1#L691), meaning it is a part of execution on the foreign chain.
So, I would add only 
```
if ccm.status == CCM_STATUS_CODE_MODULE_NOT_SUPPORTED or ccm.status == CCM_STATUS_CODE_CROSS_CHAIN_COMMAND_NOT_SUPPORTED:
    raise Exception("Improper cross-chain transfer")
Fee.payFee(FEE_CREATE_NFT)
```
Is this OK?

Reply:
I meant adding the check to command verification, so somewhere [here](https://github.com/LiskHQ/lips/blob/ca1bd3e68aaf7326de93983506eacd47f483aafa/proposals/lip-0052.md?plain=1#L646). The reason being: if an exception is raised during ccm verification, then the other chain is deemed malicious and is terminated by the Interop protocol

Reply:
Ah, get it now! Thanks!